### ISSUE TITLE
Audacity does not check permissions when the folder path contains only drive letter

### ISSUE THREAD

**Issue by Penikov** [audacity/audacity#1626]
**Describe the bug**
When a user sets only drive letter for [Preferences - Directories - Temporary files directory - Location], 
Audacity doesn't check the permissions to the directory and saves it successfully.

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to Preferences - Directories - Temporary files directory
2. Set 'C:' as the value for the 'Location' field, Click the Ok button
3. Restart Audacity

**Actual behaviour**
Audacity raises an error while launching.

**Expected behaviour**
Audacity checks directory permissions when a user clicks on the Ok button in the Preferences - - Temporary files directory

**Screenshots**
![изображение](https://user-images.githubusercontent.com/25851359/132208027-3ccd749a-227e-492d-bdfb-26d4369eed36.png)

**Additional information (please complete the following information):**
 - OS: Windows 10
 - Version Audacity 3.1.0-alpha-20210903

**Comment by petersampsonaudacity** [comment]
Testing on W10 with 3.0.4 - I get an error message in Directories prefs  when I try to set the temp directory to "C:"
![image](https://user-images.githubusercontent.com/50205984/132219872-6261230b-6a46-4756-8b29-c4d23b6f192d.png)

**Comment by Penikov** [comment]
@petersampsonaudacity Right, But if you try to fill the letter only (Without a colon), it will be valid.
Could you try to enter only the letter, please?

For example:
![изображение](https://user-images.githubusercontent.com/25851359/132253972-53974f33-37e8-486e-a634-db17784758fe.png)

**Comment by Penikov** [comment]
BTW, in my case, Audacity ate the letter with a column too.

![изображение](https://user-images.githubusercontent.com/25851359/132254097-0a1840bd-6e65-4ad5-b302-374da985f0ae.png)

But after the restart, it raises the error:
![изображение](https://user-images.githubusercontent.com/25851359/132254230-1c97e923-8b2c-4ede-85fd-d526e68782c5.png)

**Comment by Penikov** [comment]
Maybe it depends on the build version.

**Comment by petersampsonaudacity** [comment]
> @petersampsonaudacity Right, But if you try to fill the letter only (Without a colon), it will be valid.
> Could you try to enter only the letter, please?

Testing on W10 with latest alpha master audacity-win-3.1.0-alpha-20210906+1619d3d-64bit  I get
![image](https://user-images.githubusercontent.com/50205984/132260148-2ff62e83-c2ee-4b03-b377-f210adbe8b4c.png)

Then if I try to OK I het stuck in that error loop - I have to use the Cancel button.

**Comment by petersampsonaudacity** [comment]
If I use C: I get
![image](https://user-images.githubusercontent.com/50205984/132260259-e26ae715-5338-427c-8f05-6e448ea4e00c.png)

**Comment by petersampsonaudacity** [comment]
But those tests are for the Temporary directory.

As reported earlier elsewhere it's different and problemmatical for the Save and Export directories prefs:
 Directories preferences with non-existent locations results in a bad location offered for Save and Export #1625
